src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp

```diff
@@ -868,7 +968,21 @@ namespace OpenMS
         // make the feature map and assign subordinates
         OpenMS::Feature f;
         f.setUniqueId();
-        f.setMetaValue("PeptideRef", f_map.first);
+
+        // parse the identifier
+        std::string id_f;
+        try
+        {
+          std::string str = f_map.first;
+          std::string delimiter = "_";
+          id_f = str.substr(0, str.find(delimiter));
```

Comment:
String::prefix as above?

Reply:
> String::prefix as above?

We are using std::string there, do you advice to use OpenMS String instead and profit from the prefix method?

Reply:
yes

Reply:
done